base/utils/vmaas.go

```diff
@@ -0,0 +1,158 @@
+package utils
+
+import (
+	"app/base/vmaas"
+
+	"github.com/pkg/errors"
+)
+
+// Merges VMaaS responses without duplicating updates. Requires sorted input and returns sorted output.
+func MergeVMaaSResponses(vmaasDataA *vmaas.UpdatesV2Response,
+	vmaasDataB *vmaas.UpdatesV2Response) (*vmaas.UpdatesV2Response, error) {
+	if *vmaasDataA.Basearch != *vmaasDataB.Basearch {
+		return nil, errors.New("unable to merge different archs")
+	}
+
+	if *vmaasDataA.Releasever != *vmaasDataB.Releasever {
+		return nil, errors.New("release versions do not match")
+	}
+
+	if !RepoListsEqual(vmaasDataA.RepositoryList, vmaasDataB.RepositoryList) {
+		return nil, errors.New("repository lists differ")
+	}
+	if !ModuleListsEqual(vmaasDataA.ModulesList, vmaasDataB.ModulesList) {
+		return nil, errors.New("module lists differ")
+	}
+
```

Comment:
I've discussed this (different repository or module list) with others.
First thing is the data come from the same upload so the should be consistent and even if not it would be better to use one of them as a primary source then abort whole evaluation and not return anything.

So I removed this extra check.